webhook/resources/setting/validator.go

```diff
@@ -44,6 +44,19 @@ func (v *settingValidator) Create(request *admission.Request, newObj runtime.Obj
 }
 
 func (v *settingValidator) Update(request *admission.Request, oldObj runtime.Object, newObj runtime.Object) error {
+	setting := newObj.(*longhorn.Setting)
+
+	settingDef, isExist := types.GetSettingDefinition(types.SettingName(setting.Name))
+	if !isExist {
+		return werror.NewInvalidError(fmt.Sprintf("setting %s does not exist", setting.Name), "metadata.name")
+	}
+	existingSetting := oldObj.(*longhorn.Setting)
+	_, isFromLHOld := existingSetting.Annotations[types.GetLonghornLabelKey(types.UpdateSettingFromLonghorn)]
+	_, isFromLH := setting.Annotations[types.GetLonghornLabelKey(types.UpdateSettingFromLonghorn)]
+	if settingDef.ReadOnly && !isFromLHOld && !isFromLH {
```

Comment:
Why do we need to check the annotation of the old setting? If the new update has an annotation, it simply means we need to update it, right?